### PR TITLE
golang example: use go build command to compile go source code.

### PR DESCRIPTION
**Pull request by doujiang24**
In the golang world, we usually use the go build command, instead of bazel.

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by doujiang24** [comment]
/assign @phlax 
Thanks!

File: examples/golang/docker-compose-go.yaml [hunk 1]
```diff
@@ -2,10 +2,13 @@ services:
   go_plugin_compile:
     build:
       context: ../shared/build
```
Comment (phlax) [inline examples/golang/docker-compose-go.yaml:4]: should we just use a golang container in this case ?

there are some recipes already using it check `shared/golang` and any examples that link to it

File: examples/golang/simple/go.mod [hunk 2]
```diff
@@ -0,0 +1,24 @@
+module example.envoyproxy.com/simple
```
[the remark this answers is not in the document]
Comment (doujiang24) [reply examples/golang/simple/go.mod]: done, thank you.

**Review by phlax — APPROVED** [review]
lgtm, thanks @doujiang24

**Pull request merged**